File: editDistance.py
class Solution:
    # @return an integer
    def minDistance(self, word1, word2):
    	if len(word2) * len(word1) == 0:
    		return max(len(word1), len(word2))
        metrics = [["invalid" for j in range(len(word2) + 1)] for i in range(len(word1) + 1)]
        metrics[0] = range(len(word2) + 1)
        for i in range(len(metrics)):
        	metrics[i][0] = i
        #print metrics
        for i in range(1, len(metrics)):
        	for j in range(1, len(metrics[i])):
        		if word1[i-1] == word2[j-1]:
        			metrics[i][j] = min(metrics[i-1][j-1], metrics[i-1][j] + 1, metrics[i][j-1] + 1)
        		else:
        			metrics[i][j] = min(metrics[i-1][j-1], metrics[i-1][j], metrics[i][j-1]) + 1
        # for i in metrics:
        # 	print i
        return metrics[len(word1)][len(word2)]

s = Solution()
print s.minDistance('plasma', 'altruism')

File: JAVA/editDistance.java
public class Solution {
    public int minDistance(String word1, String word2) {
        int[] lastRow = new int[word2.length()+1];
        int[] currentRow = new int[word2.length()+1], temp;
        for (int i=0;i<word2.length()+1;i++)
            currentRow[i] = i;
        int count = 1;
        while(count < word1.length()+1){
            temp = lastRow;
            lastRow = currentRow;
            currentRow = temp;
            currentRow[0] = count;
            for(int j=1;j<word2.length()+1;j++)
                currentRow[j] = Math.min(Math.min(currentRow[j-1]+1, lastRow[j]+1), lastRow[j-1]+(word1.charAt(count-1) == word2.charAt(j-1)? 0:1));
            count++;
        }
        return currentRow[word2.length()];
    }
}